### ISSUE TITLE
explicit_iter_loop should warn &[T].iter()

### ISSUE THREAD

**Issue by malbarbo** [rust-lang/rust-clippy#1518]
```rust
fn main() {
    let s = &[1, 2, 3, 4];
    for a in s.iter() { // .iter() is not necessary
        println!("{:?}", a);
    }
}
```

**Comment by llogiq** [comment]
As long as `s` `Deref`s to a `slice`.

**Comment by retep998** [comment]
This also applies to `for foo in bar.iter()` when `bar: &Vec<T>`.
